pycqed/measurement/calibration/single_qubit_gates.py

```diff
@@ -233,48 +321,255 @@ def __init__(self, task_list, sweep_points=None, **kw):
             self.preprocessed_task_list, self.sweep_block, **kw)
 
     def resolve_lo_sweep_points(self, freq_sp_suffix='freq', **kw):
-        all_freqs = self.sweep_points.get_sweep_params_property('values', 1,
-                                                                'all')
+        """
+        :param freq_sp_suffix: (str, default 'freq') To identify the
+            frequency sweep parameter, this string specifies a suffix that is
+            contained at the end of the name of the frequency sweep parameter.
+        :param kw: keyword arguments, see class docstring
+        """
+        all_freqs = np.array(
+            self.sweep_points.get_sweep_params_property('values', 1, 'all'))
         if np.ndim(all_freqs) == 1:
             all_freqs = [all_freqs]
         all_diffs = [np.diff(freqs) for freqs in all_freqs]
-        assert all([np.mean(abs(diff - all_diffs[0]) / all_diffs[0]) < 1e-10
-                    for diff in all_diffs]), \
+        assert all([len(d) == 0 for d in all_diffs]) or \
+            all([np.mean(abs(diff - all_diffs[0]) / all_diffs[0]) < 1e-10
+                 for diff in all_diffs]), \
```

Comment:
To my understanding, np.isclose compares two arrays, but here I need n arrays. So it would maybe not simplify the code that much. But I will keep np.isclose in mind in the future.